### PR TITLE
[FIX] Message search results (@mentions) do not have any date indication in the results

### PR DESCRIPTION
**Pull request by sudipt1999**
Please review the changes ! Now I am showing full date `Hr:Min Month Day, Year` kindly cross check. 
Actual link of issue https://github.com/WideChat/Rocket.Chat.Android/issues/337

### CODE REVIEW

**Comment by ear-dev** [comment]
@sudipt1999 I built this, but I'm getting an error message each time I try to open a chatroom.... any ideas?

05-17 11:37:16.731 17727-17727/chat.rocket.android.dev E/ChatRoomPresenter$loadMessages: org.threeten.bp.DateTimeException: Unable to extract value: class org.threeten.bp.LocalDateTime
        at org.threeten.bp.format.DateTimePrintContext.getValue(DateTimePrintContext.java:258)
        at org.threeten.bp.format.DateTimeFormatterBuilder$ZoneTextPrinterParser.print(DateTimeFormatterBuilder.java:3362)
        at org.threeten.bp.format.DateTimeFormatterBuilder$CompositePrinterParser.print(DateTimeFormatterBuilder.java:1995)
        at org.threeten.bp.format.DateTimeFormatterBuilder$LocalizedPrinterParser.print(DateTimeFormatterBuilder.java:3751)
        at org.threeten.bp.format.DateTimeFormatterBuilder$CompositePrinterParser.print(DateTimeFormatterBuilder.java:1995)
        at org.threeten.bp.format.DateTimeFormatter.formatTo(DateTimeFormatter.java:1385)
        at org.threeten.bp.format.DateTimeFormatter.format(DateTimeFormatter.java:1359)
        at org.threeten.bp.chrono.ChronoLocalDateTime.format(ChronoLocalDateTime.java:263)
        at org.threeten.bp.LocalDateTime.format(LocalDateTime.java:1828)
        at DateTimeHelper.getTime(DateTimeHelper.kt:69)
        at chat.rocket.android.chatroom.uimodel.UiModelMapper.getTime(UiModelMapper.kt:535)
        at chat.rocket.android.chatroom.uimodel.UiModelMapper.access$getTime(UiModelMapper.kt:56)
        at chat.rocket.android.chatroom.uimodel.UiModelMapper$mapMessage$2.invokeSuspend(UiModelMapper.kt:443)
        at chat.rocket.android.chatroom.uimodel.UiModelMapper$mapMessage$2.invoke(Unknown Source:33)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:146)
        at kotlinx.coroutines.BuildersKt.withContext(Unknown Source:1)
        at chat.rocket.android.chatroom.uimodel.UiModelMapper.mapMessage(UiModelMapper.kt:441)
        at chat.rocket.android.chatroom.uimodel.UiModelMapper$translate$2.invokeSuspend(UiModelMapper.kt:143)
        at chat.rocket.android.chatroom.uimodel.UiModelMapper$translate$2.invoke(Unknown Source:33)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:146)
        at kotlinx.coroutines.BuildersKt.withContext(Unknown Source:1)
        at chat.rocket.android.chatroom.uimodel.UiModelMapper.translate(UiModelMapper.kt:124)
        at chat.rocket.android.chatroom.uimodel.UiModelMapper$map$4.invokeSuspend(UiModelMapper.kt:97)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:233)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:742)

**Comment by sudipt1999** [comment]
current


Current changes!

**Comment by jaytat0** [comment]
@sudipt1999 I don't believe this addresses what the initial issue was intending to get fixed (https://github.com/WideChat/Rocket.Chat.Android/issues/337), which is that when you look at the screenshot from that: 
image
What's showing are the results of a search for "bulls". But a user has no way of knowing what date those search results are from. So what's needed is a date indication for search results, and the same applies for @mentions.

I don't believe that the "today" change is needed. Once date indicators are put in properly, a result  from a year ago will have the year showing  ("May 20, 2018") so it won't be confused with today.

File: app/src/main/java/chat/rocket/android/app/DateTimeHelper.kt [hunk 1]
```diff
@@ -37,7 +37,7 @@ object DateTimeHelper {
     fun getDate(localDateTime: LocalDateTime, context: Context): String {
         val localDate = localDateTime.toLocalDate()
         return when (localDate) {
-            today -> formatLocalTime(localDateTime.toLocalTime())
+            today -> "Today " + formatLocalTime(localDateTime.toLocalTime())
```
Comment (hfeky) [inline app/src/main/java/chat/rocket/android/app/DateTimeHelper.kt]: Please avoid using hardcoded strings. Add that to strings resource files for all languages.

**Review by hfeky — CHANGES_REQUESTED** [review]

File: app/src/main/java/chat/rocket/android/app/DateTimeHelper.kt [hunk 2]
```diff
@@ -66,6 +67,9 @@ object DateTimeHelper {
      */
     fun getTime(localDateTime: LocalDateTime): String {
         val formatter = DateTimeFormatter.ofLocalizedTime(FormatStyle.SHORT)
+        if(isSearching){
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/app/DateTimeHelper.kt]: Make sure to follow our coding style.

**Comment by philipbrito** [comment]
@sudipt1999 Are you going to make changes here based on the comments done by @jaytat0 ?

**Comment by sudipt1999** [comment]
@filipedelimabrito yes sir changes are already made for that! As you mentioned to change the coding style will do that today.

**Comment by sudipt1999** [comment]
@filipedelimabrito Please review the pr.

File: app/src/main/java/chat/rocket/android/app/DateTimeHelper.kt [hunk 3]
```diff
@@ -66,7 +67,11 @@ object DateTimeHelper {
      */
     fun getTime(localDateTime: LocalDateTime): String {
         val formatter = DateTimeFormatter.ofLocalizedTime(FormatStyle.SHORT)
-        return localDateTime.toLocalTime().format(formatter).toString()
+        if (isSearching) {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/app/DateTimeHelper.kt:70]: The logic here could be improved in more that one way.

1 - You could make the return lifted out of if.
2 - If you are searching you don't need to have the formatter variable on the top.

File: app/src/main/java/chat/rocket/android/app/DateTimeHelper.kt [hunk 4]
```diff
@@ -15,7 +15,8 @@ object DateTimeHelper {
     private val today = LocalDate.now()
     private val yesterday = today.minusDays(1)
     private val lastWeek = today.minusWeeks(1)
-
+    var isSearching = false
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/app/DateTimeHelper.kt:18]: You can make it working without declaring it, this way you could also avoid creating a new function here.

**Review by philipbrito — CHANGES_REQUESTED** [review]

**Review by philipbrito — APPROVED** [review]
Code improvement will be addressed by this PR: https://github.com/RocketChat/Rocket.Chat.Android/pull/2390

Thanks for sending this one!

**Pull request merged**